src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -0,0 +1,213 @@
+package org.ergoplatform.nodeView.wallet
+
+import akka.actor.Actor
+import io.iohk.iodb.ByteArrayWrapper
+import org.ergoplatform.modifiers.mempool.{ErgoTransaction, UnsignedErgoTransaction}
+import org.ergoplatform.nodeView.history.ErgoHistory.Height
+import org.ergoplatform._
+import org.ergoplatform.modifiers.ErgoFullBlock
+import org.ergoplatform.nodeView.state.ErgoStateContext
+import org.ergoplatform.nodeView.wallet.BoxCertainty.Uncertain
+import org.ergoplatform.settings.ErgoSettings
+import org.ergoplatform.utils.AssetUtils
+import scorex.core.utils.ScorexLogging
+import scorex.crypto.authds.ADDigest
+import scorex.crypto.hash.Digest32
+import sigmastate.interpreter.ContextExtension
+import sigmastate.{AvlTreeData, Values}
+
+import scala.collection.Map
+import scala.collection.mutable
+import scala.util.{Failure, Random, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+case class BalancesSnapshot(height: Height, balance: Long, assetBalances: Map[ByteArrayWrapper, Long])
+
+
+class ErgoWalletActor(settings: ErgoSettings) extends Actor with ScorexLogging {
```

Comment:
Please add comments, what are the goals of this class. 
Why do we need separate actor for Wallet?